### PR TITLE
TST: Add infrastructure for flake8 tests

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Adding the flake8 option to the testing suite.  The basic structure is similar to that used in the new packages (e.g., pysatModels).  Since scrubbing the entire code is a large process, exceptions have been added for many of the current flags, with the intent to remove these as the code is generally scrubbed over a series of smaller style pulls.

A handful of changes (mostly whitespace) have been added here for files with relatively few flake8 errors. Currently 39 of 82 files have exceptions flagged.  These will be removed over the course of several smaller pulls.

Note that the pytest settings have been moved from pytest.ini to setup.cfg.

## Type of change

- Testing fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Changes to pytest config.

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: pysat/instruments/cosmic_gps.py [hunk 1]
```diff
@@ -132,8 +132,9 @@ def list_files(tag=None, sat_id=None, data_path=None, format_str=None):
         minute = np.array(stored['minute'])
         try:
             uts = hour*3600.0 + minute*60.0
-        except (TypeError, UFuncTypeError) as err:
-            raise TypeError('unable to construct time from filename\n{:}'.format(err))
+        except TypeError as err:
+            raise TypeError(' '.join(('unable to construct time from',
+                                      'filename\n{:}'.format(str(err)))))
```
Comment (aburrell) [inline pysat/instruments/cosmic_gps.py:137]: This change can cause problems.  Do we know that the `UFuncTypeError` won't be raised?

Comment (jklenzing) [reply pysat/instruments/cosmic_gps.py:137]: `UFuncTypeError` will be caught by `TypeError`.  Checked into the rationale for this in #442.  `UFuncTypeError` is raised when we try to multiply a numpy array of strings by a number.  Test code below.

```
import numpy as np
hour = np.array([20, 21, 22])  
minute = np.array(['1', '2', '3']) 
try: 
    uts = hour*3600.0 + minute*60.0 
except TypeError: 
    print('Everything is fine') 
```

File: setup.cfg [hunk 2]
```diff
@@ -1,3 +1,7 @@
+[report]
```
Comment (aburrell) [inline setup.cfg:1]:
```suggestion
[coverage:report]
```

File: .travis.yml [hunk 3]
```diff
@@ -33,14 +33,15 @@ install:
...
   - python setup.py install
 
 # command to run tests
 script:
- - pytest -vs --cov=pysat/
+ - pytest -vs --flake8 --cov=pysat/
 
 after_success:
- - coveralls
+ - coveralls --rcfile
```
Comment (aburrell) [inline .travis.yml:47]:
```suggestion
 - coveralls --rcfile=setup.cfg
```

**Comment by rstoneback** [comment]
Thanks @jklenzing and @aburrell

**Comment by jklenzing** [comment]
Not sure why tests show up as pending here.  Completed and passing in travis.

**Pull request merged**